allennlp/modules/conditional_random_field.py

```diff
@@ -278,7 +278,10 @@ def forward(self,
 
         return torch.sum(log_numerator - log_denominator)
 
-    def viterbi_tags(self, logits: torch.Tensor, mask: torch.Tensor) -> List[List[int]]:
+    def viterbi_tags(self,
+                     logits: torch.Tensor,
+                     mask: torch.Tensor,
+                     keep_scores: bool = False) -> Union[List[List[int]], Tuple[List[List[int]], List[float]]]:
```

Comment:
I don't like functions that return union types, they make for a muddy API.

I would rather just change the function to return (say) `List[Tuple[List[int], float]]` always, and then modify the handful of downstream consumers to behave gracefully and ignore the scores if they don't want them.